File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/exams-management/models/exam-list-item.dto.ts
import { dropdownOptionsModel } from '../../../../../shared/models/dropdown-options.model';

export interface ExamListItemDto {
  id: string;
  examNo: string;
  jobTitle: string;
  specializations: string[];
  totalQuestions: number;
  totalDurationMinutes: number;
  status: dropdownOptionsModel;
  createdDate: string;
  lastUpdated: string;
}

export interface ExamFilters {
  pageNumber: number;
  pageSize: number;
  language: string;
  search?: string;
  specializationId?: string;
  statusId?: string;
  createdFrom?: string;
  createdTo?: string;
}

File: src/Application.Operation/Features/Employee/Exams/DTOs/ExamListItemDto.cs
using Tawtheef.Application.Common.Models;

namespace Application.Operation.Features.Employee.Exams.DTOs;

public sealed record ExamListItemDto
{
    public Guid Id { get; init; }
    public required string ExamNo { get; init; }
    public required string JobTitle { get; init; }
    public required List<string> Specializations { get; init; }
    public int TotalQuestions { get; init; }
    public int TotalDurationMinutes { get; init; }
    public required DropdownOptions Status { get; init; }
    public DateTime CreatedDate { get; init; }
    public DateTime LastUpdated { get; init; }
}
